File: trueshot-backend/src/main/java/com/ongtangco/trueshot/model/Order.java
package com.ongtangco.trueshot.model;

import com.ongtangco.trueshot.enums.OrderStatus;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
public class Order {
    private Integer id;
    private String buyerName;
    private String buyerEmail;
    private String buyerPhone;
    private OrderStatus status;
    private BigDecimal totalAmount;
    private String specialInstructions;
    private LocalDateTime createdAt;
    private LocalDateTime lastUpdated;
    private List<OrderItem> items = new ArrayList<>();
    private TransactionDetails transactionDetails;
}

File: trueshot-frontend/src/app/models/order.ts
export interface OrderItem {
  id?: number;
  orderId?: number;
  productId?: number;
  productName?: string;
  productBrand?: string;
  productCondition?: string;
  productImageUrl?: string;
  unitPrice?: number;
  quantity?: number;
  lineTotal?: number;
  status?: string;
}

export interface Order {
  id?: number;
  buyerName?: string;
  buyerEmail?: string;
  buyerPhone?: string;
  status?: string;
  totalAmount?: number;
  specialInstructions?: string;
  createdAt?: string;
  lastUpdated?: string;
  items?: OrderItem[];
}
